initializr-generator-spring/src/test/java/io/spring/initializr/generator/spring/build/gradle/GradleProjectGenerationConfigurationTests.java

```diff
@@ -118,7 +118,9 @@ void buildDotGradleIsContributedWhenGeneratingGradleProject() {
 				"",
 				"group = 'com.example'",
 				"version = '0.0.1-SNAPSHOT'",
-				"sourceCompatibility = '11'",
+				"java {",
+				"    sourceCompatibility = JavaVersion.VERSION_11",
```

Comment:
I prefer that we continue to use a String for the version as it's more concise and, IMO, it reads better.

Reply:
@snicoll @wilkinsona, I read in https://docs.gradle.org/8.2-rc-1/dsl/org.gradle.api.plugins.JavaPluginExtension.html#org.gradle.api.plugins.JavaPluginExtension:sourceCompatibility
that source compatibility is going to be like java version constant. That is why I added it. So should we continue with String for version or add Java version constant ?

Reply:
Gradle will perform the `Object -> JavaVersion` conversion as it's https://docs.gradle.org/8.2-rc-1/javadoc/org/gradle/api/plugins/JavaPluginExtension.html#setSourceCompatibility-java.lang.Object- that will be called.

Reply:
Thanks. I understand now.